modules/apps/transfer/ibc_module_test.go

```diff
@@ -378,27 +363,28 @@ func (suite *TransferTestSuite) TestOnChanUpgradeTry() {
 			err := path.EndpointA.ChanUpgradeInit()
 			suite.Require().NoError(err)
 
+			counterpartyUpgrade = path.EndpointA.GetChannelUpgrade()
+
 			tc.malleate()
 
-			err = path.EndpointB.ChanUpgradeTry()
+			module, _, err := suite.chainB.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainB.GetContext(), types.PortID)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainB.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			version, err := cbs.OnChanUpgradeTry(
+				suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID,
+				counterpartyUpgrade.Fields.Ordering, counterpartyUpgrade.Fields.ConnectionHops, counterpartyUpgrade.Fields.Version,
+			)
 
 			expPass := tc.expError == nil
 			if expPass {
 				suite.Require().NoError(err)
-
-				upgrade := path.EndpointB.GetChannelUpgrade()
-				suite.Require().Equal(upgradePath.EndpointB.ConnectionID, upgrade.Fields.ConnectionHops[0])
+				suite.Require().Equal(types.Version, version)
 			} else {
-				// NOTE: application callback failure in OnChanUpgradeTry results in an ErrorReceipt being written to state signaling for cancellation
-				if expUpgradeError, ok := tc.expError.(*channeltypes.UpgradeError); ok {
-					errorReceipt, found := suite.chainB.GetSimApp().GetIBCKeeper().ChannelKeeper.GetUpgradeErrorReceipt(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
-					suite.Require().True(found)
-					suite.Require().Equal(expUpgradeError.GetErrorReceipt(), errorReceipt)
-				} else {
-					// NOTE: application callback is not reached and instead an error is returned directly to the client
-					suite.Require().Error(err)
-					suite.Require().ErrorIs(err, tc.expError)
-				}
+				suite.Require().Error(err)
+				suite.Require().ErrorIs(err, tc.expError)
```

Comment:
This whole `expPass` block could be removed and just have a single:

```go
suite.Require().ErrorIs(err, tc.expError)
```

But I'll just keep this as is for now as its aligns with all other test structure. Happy to change later if anyone really feels strong about it.

Reply:
:muscle:  FEELING STRONK :muscle: 

nah, fine either way, didn't wake up today feeling overscrupulous